### ISSUE TITLE
Error "./run_jupyter.sh: no such file or directory"

### ISSUE THREAD

**Issue by bigredbug47** [hamelsmu/Docker_Tutorial#6]
Hi,
I read your medium article and following it for installing, but when I run code:
`sudo docker run -it --name container1 --net=host -v ~/docker_files/:/ds tutorial~`

It returns this error:
`docker: Error response from daemon: OCI runtime create failed: container_linux.go:348: starting container process caused "exec: \"./run_jupyter.sh\": stat ./run_jupyter.sh: no such file or directory": unknown.`

I had download file "run_jupyter.sh" from your repo, put in both root and current directory but nothing works.

Thks alot.

**Comment by bigredbug47** [comment]
[UPDATED]
When I change last line in Dockerfile
from
`CMD ['./run_jupyter.sh']`
to
`CMD ['/bin/bash']`

It works and return me to ds folder in cointainer1:
`root@docker`
but I realize that the file "run_jupyter.sh" that not exists. But why it happened? I recheck in Dockerfile and the code ADD run_jupyter file is still there
`ADD run_jupyter.sh /ds/run_jupyter.sh`

**Comment by beatthat** [comment]
I have the exact same issue. Any new insight on this?

Here's my env info:

$ uname -a
Darwin harleywinks 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64

$ docker --version
Docker version 18.06.1-ce, build e68fc7a

**Comment by LorisMarini** [comment]
Solved by changing the content of run_container.sh from:

docker run -it --net=host -v ~/docker_files/:/ds tutorial

to:

CONT_LOC=/path/to/basic/tutorial/directory/
docker run -it --net=host -v $CONT_LOC:/ds tutorial

Hope this helps.

**Comment by arnabid** [comment]
@hamelsmu The "RUN chmod +x /ds/run_jupyter.sh" instruction in the Dockerfile does not work. Any ideas?

found this reference: https://github.com/moby/moby/issues/34819

**Comment by celsofranssa** [comment]
You can change from CMD `['./run_jupyter.sh']` to `CMD ['/bin/bash']` like @bigredbug47 said and after creating the image from the new Dockerfile, run it using:

`docker run -it --name jupyter-container --net=host -v ~/docker_files/:/ds tutorial`

when the container comes to life, just run inside it:

`jupyter notebook --no-browser --allow-root --port=7745 --NotebookApp.token='tutorial'`

Finally, go to the `http://localhost:7745/` and see the magic :D

I hope this is useful for you.

**Comment by ckvtvm-arch** [comment]
> [UPDATED]
> When I change last line in Dockerfile
> from
> `CMD ['./run_jupyter.sh']`
> to
> `CMD ['/bin/bash']`
> 
> It works and return me to ds folder in cointainer1:
> `root@docker`
> but I realize that the file "run_jupyter.sh" that not exists. But why it happened? I recheck in Dockerfile and the code ADD run_jupyter file is still there
> `ADD run_jupyter.sh /ds/run_jupyter.sh`

I am facing the same issue - Entrypoint and CMD line in dockerfile for run_jupyter.sh says no such file or directory exits though it is present in the notebooks wokr directory . please help

RUN chown -R 1001:0 /opt/app-root && chmod -R og+rwx /opt/app-root

#RUN ln -s /usr/bin/python3

# Set up our notebook config.
COPY jupyter_notebook_config.py $HOME/.jupyter/

# Copy sample notebooks.
COPY notebooks /notebooks
COPY run_jupyter.sh /notebooks
RUN chmod -R og+rwx /notebooks
#RUN chmod +x notebooks /run_jupyter.sh 

# Jupyter has issues with being run directly:
#   https://github.com/ipython/ipython/issues/7062
# We just add a little wrapper script.
USER 1001
WORKDIR /notebooks
CMD ["/bin/bash"]
#CMD ["/run_jupyter.sh"]
#CMD ["/run_jupyter.sh", "--allow-root"]
#ENTRYPOINT "/run_jupyter.sh"

**Comment by hamelsmu** [comment]
Thanks I added these instructions https://github.com/hamelsmu/Docker_Tutorial/blob/master/basic_tutorial/README.md

And also made some changes to the Dockerfile.  Should work now

**Comment by ckvtvm-arch** [comment]
> Thanks I added these instructions https://github.com/hamelsmu/Docker_Tutorial/blob/master/basic_tutorial/README.md
> 
> And also made some changes to the Dockerfile. Should work now

thanks .

Step 21/22 : RUN chmod +x /run_jupyter.sh
 ---> Running in e1c5393b012b
chmod: cannot access '/run_jupyter.sh': No such file or directory
The command '/bin/sh -c chmod +x /run_jupyter.sh' returned a non-zero code: 1

**Comment by ckvtvm-arch** [comment]
full code as given below:
# random UID
RUN chown -R 1001:0 /opt/app-root && chmod -R og+rwx /opt/app-root

#RUN ln -s /usr/bin/python3

# Set up our notebook config.
COPY jupyter_notebook_config.py $HOME/.jupyter/

# Copy sample notebooks.
COPY notebooks /notebooks
COPY run_jupyter.sh /notebooks
RUN chmod -R og+rwx /notebooks
#RUN chmod +x notebooks /run_jupyter.sh 

# Jupyter has issues with being run directly:
#   https://github.com/ipython/ipython/issues/7062
# We just add a little wrapper script.
USER 1001
WORKDIR /notebooks
#CMD ["/bin/bash"]
RUN chmod +x /run_jupyter.sh
CMD ["/run_jupyter.sh"]
#CMD ["/run_jupyter.sh", "--allow-root"]
#ENTRYPOINT "/run_jupyter.sh"

**Comment by hamelsmu** [comment]
Did you pull the new changes?

**Comment by ckvtvm-arch** [comment]
> Did you pull the new changes?
I am using the below dockerfile which is almost similar, but not the same. But the issue is referred here , hence trying to get a direction.

#FROM centos:latest
FROM centos/s2i-base-centos7

MAINTAINER Nick Harvey <nickharveyonline@gmail.com>

# TensorBoard
EXPOSE 6006
# IPython
EXPOSE 8888

LABEL io.k8s.description="An open-source software library for Machine Intelligence" \
      io.k8s.display-name="Tensorflow 0.12" \
      io.openshift.expose-services="6006:http" \
      io.openshift.expose-services="8888:http" \
      io.openshift.tags="builder,python,python35,rh-python35,tensorflow"

#Update centos 
RUN yum update -y

#Install Dependencies
RUN yum groupinstall 'Development Tools' -y
RUN yum install -y epel-release
RUN yum install -y \
        curl \
        freetype-devel \
        libpng12-devel \
        pkgconfig \
        python \
        python-devel \
        rsync \
        unzip \
        nss_wrapper \
        gettext \
        python-pip \
        atlas \
        atlas-devel \
        gcc-gfortran \
        openssl-devel \ 
        libffi-devel \
        && \
        yum clean all -y

#RUN curl --insecure -O https://bootstrap.pypa.io/get-pip.py && \
 #   python get-pip.py && \
 #  rm get-pip.py
RUN INSTALL_PKGS="python3 python3-devel" && \
    curl  --insecure https://raw.githubusercontent.com/cloudrouter/centos-repo/master/CentOS-Base.repo -o /etc/yum.repos.d/CentOS-Base.repo && \
    curl  --insecure http://mirror.centos.org/centos-7/7/os/x86_64/RPM-GPG-KEY-CentOS-7 -o /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 && \
   # DISABLES="--disablerepo=cbs-paas7-openshift-multiarch-el7-build" && \
     yum install -y $INSTALL_PKGS 
#USER 1001
RUN pip3 install  --trusted-host files.pythonhosted.org --trusted-host pypi.org --trusted-host pypi.python.org -U pip setuptools wheel
RUN pip3 install --trusted-host=pypi.org --trusted-host=files.pythonhosted.org \ 
        ipykernel \
        jupyter \ 
        matplotlib \
        numpy \
        scipy \
        sklearn \
        Pillow \
        && \
    python3 -m ipykernel.kernelspec

# --- DO NOT EDIT OR DELETE BETWEEN THE LINES --- #
# These lines will be edited automatically by parameterized_docker_build.sh. #
# COPY _PIP_FILE_ /
# RUN pip --no-cache-dir install /_PIP_FILE_
# RUN rm -f /_PIP_FILE_

# Install TensorFlow CPU version from central repo
RUN pip3 --no-cache-dir install --trusted-host=pypi.org --trusted-host=files.pythonhosted.org --upgrade tensorflow 
   # https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.7.1-cp34-none-linux_x86_64.whl
    #http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.12.1-cp27-none-linux_x86_64.whl
#http://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-0.0.0-cp27-none-linux_x86_64.whl
# --- ~ DO NOT EDIT OR DELETE BETWEEN THE LINES --- #

# original code - start
# In order to drop the root user, we have to make some directories world
# writable as OpenShift default security model is to run the container under
# random UID
RUN chown -R 1001:0 /opt/app-root && chmod -R og+rwx /opt/app-root

#RUN ln -s /usr/bin/python3

# Set up our notebook config.
COPY jupyter_notebook_config.py $HOME/.jupyter/

# Copy sample notebooks.
COPY notebooks /notebooks
COPY run_jupyter.sh /notebooks
RUN chmod -R og+rwx /notebooks
#RUN chmod +x notebooks /run_jupyter.sh 

# Jupyter has issues with being run directly:
#   https://github.com/ipython/ipython/issues/7062
# We just add a little wrapper script.
USER 1001
WORKDIR /notebooks
#CMD ["/bin/bash"]
RUN chmod +x /run_jupyter.sh
CMD ["/run_jupyter.sh"]
#CMD ["/run_jupyter.sh", "--allow-root"]
#ENTRYPOINT "/run_jupyter.sh"

**Comment by hamelsmu** [comment]
Ask on stack overflow

**Comment by ckvtvm-arch** [comment]
ok sure thanks
